src/Diagnostics.DataProviders/DataProviders/ChangeAnalysisDataProvider.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Threading.Tasks;
+using Diagnostics.DataProviders.DataProviderConfigurations;
+using Diagnostics.ModelsAndUtils.Models.ChangeAnalysis;
+using Diagnostics.DataProviders.Interfaces;
+
+namespace Diagnostics.DataProviders
+{
+    public class ChangeAnalysisDataProvider : DiagnosticDataProvider, IChangeAnalysisDataProvider
+    {
+        private ChangeAnalysisDataProviderConfiguration dataProviderConfiguration;
+
+        private ChangeAnalysisClient changeAnalysisClient;
+
+        private string dataProviderRequestId;
+
+        private KustoDataProvider kustoDataProvider;
+
+        public ChangeAnalysisDataProvider(OperationDataCache cache, ChangeAnalysisDataProviderConfiguration configuration, KustoDataProviderConfiguration kustoConfig, string requestId, string clientObjectId, string principalName) : base(cache)
+        {
+            dataProviderConfiguration = configuration;
+            dataProviderRequestId = requestId;
+            changeAnalysisClient = new ChangeAnalysisClient(clientObjectId, principalName);
+            kustoDataProvider = new KustoDataProvider(cache, kustoConfig, requestId);
+        }
+
+        /// <summary>
+        /// Get all change sets for the given arm resource uri.
+        /// </summary>
+        /// <param name="armResourceUri">ARM Resource URI.</param>
+        /// <param name="startTime">Start time of time range.</param>
+        /// <param name="endTime">End time of the time range.</param>
+        /// <returns>List of changesets.</returns>
+        public async Task<List<ChangeSetResponseModel>> GetChangeSetsForResource(string armResourceUri, DateTime startTime, DateTime endTime)
+        {
```

Comment:
Detector owners will be calling this method, I suggest checking for null or empty with `isNullOrWhitespace` for the `armResourceUri`.

Reply:
Does the change analysis API have any limitations on starttime and endtime. Like do they only keep changelog information for the last 30 days? If we don't know that yet then its fine we can just move forward.

Reply:
There is a limitation I believe. I will check again with their team to find out what the range is and update the method to validate for that.